src/ovs/aca_ovs_programmer.cpp

```diff
@@ -34,51 +39,78 @@ ACA_OVS_Programmer &ACA_OVS_Programmer::get_instance()
   return instance;
 }
 
-int ACA_OVS_Programmer::setup_bridges()
+int ACA_OVS_Programmer::setup_ovs_bridges_if_need()
 {
-  ACA_LOG_DEBUG("ACA_OVS_Programmer::setup_bridges ---> Entering\n");
+  ACA_LOG_DEBUG("ACA_OVS_Programmer::setup_ovs_bridges_if_need ---> Entering\n");
 
   ulong not_care_culminative_time;
   int overall_rc = EXIT_SUCCESS;
 
-  // TODO: confirm OVS is up and running, if not, try to start it
+  // -----critical section starts-----
+  // (exclusive access to br-int and br-tun creation status signaled by locking setup_ovs_bridges_mutex):
+  setup_ovs_bridges_mutex.lock();
```

Comment:
will this potentially create contention for concurrent port create?

Reply:
Let's break it down to multiple situations:

* for most case (case 1), the bridges are already created, it will be a quick and needed check within the mutex and return.
* for the less likely case (case 2), the bridges are not there so we need to create them, we don't want other port create to go in to check at this time so mutex is needed. Other port create will have to wait until the bridges are fully created for the right reason.